### PR TITLE
Pepi

### PR DESCRIPTION
**Pull request by DonnuTs94**
adding controller , service and routes for get all data product with pagination and search

### CODE REVIEW

File: app/controllers/productController.js [hunk 1]
```diff
@@ -40,6 +44,42 @@ const productController = {
...
       })
     }
+  },
+  getAllProduct: async (req, res) => {
+    try {
+      const { product, category, page } = req.query
+
+      const pageNumber = page ? parseInt(page, 10) : 1
+
+      if (isNaN(pageNumber) || pageNumber < 1) {
```
Comment (Inun572) [inline app/controllers/productController.js]: kurang convert to number

Comment (DonnuTs94) [reply app/controllers/productController.js]: itu hasilnya udh di tabrak di bawah mi

Comment (DonnuTs94) [reply app/controllers/productController.js]: ketika dia masuk string udh di handle langsung

Comment (Inun572) [reply app/controllers/productController.js]: oh oke, udah pake parseInt ya

**Review by Inun572 — APPROVED** [review]
oke

**Pull request merged**